test/common/http/http2/codec_impl_test_util.h

```diff
@@ -59,10 +59,21 @@ class TestClientConnectionImpl : public ClientConnectionImpl, public TestCodecSe
   TestClientConnectionImpl(Network::Connection& connection, Http::ConnectionCallbacks& callbacks,
                            Stats::Scope& scope,
                            const envoy::config::core::v3::Http2ProtocolOptions& http2_options,
-                           uint32_t max_request_headers_kb, uint32_t max_request_headers_count)
+                           uint32_t max_request_headers_kb, uint32_t max_request_headers_count,
+                           bool test_only_session = false)
       : ClientConnectionImpl(connection, callbacks, scope, http2_options, max_request_headers_kb,
-                             max_request_headers_count) {}
+                             max_request_headers_count, test_only_session) {}
+
   nghttp2_session* session() { return session_; }
+
+  // Submits an H/2 METADATA frame to the peer.
+  // Returns true on success, false otherwise.
+  virtual bool submitMetadata(const MetadataMapVector& mm_vector, int32_t stream_id) {
```

Comment:
any reason to not just make this
submitMetadata(const MetadataMapVector&, int32_t)
and remove the UNREFERENCED_PARAMETERS ?

Reply:
This one get overlooked?

Reply:
I tend to prefer more descriptive declarations for API clarity and I think having variable names helps with that. If you think this creates unnecessary clutter, I can remove it.

Reply:
Gotcha.  I disagree and I think it's not common but I'll buy it's a style choice :-)